Remove the word "mongrel" from documents

tricknotes opened the pull request:
### Summary

Currently [mongrel](https://github.com/mongrel/mongrel) is not maintained.
And it couldn't be built with any Ruby versions that supported by Rails.

It is reasonable to remove the word "mongrel" in order to avoid confusion from newcomer.

vipulnsward commented on activesupport/lib/active_support/backtrace_cleaner.rb:
```diff
@@ -14,7 +14,7 @@ module ActiveSupport
   #   bc = BacktraceCleaner.new
   #   bc.add_filter   { |line| line.gsub(Rails.root.to_s, '') } # strip the Rails.root prefix
-  #   bc.add_silencer { |line| line =~ /mongrel|rubygems/ } # skip any lines from mongrel or rubygems
+  #   bc.add_silencer { |line| line =~ /my_library|rubygems/ } # skip any lines from my_library or rubygems
```
can you change to puma

vipulnsward commented on activesupport/lib/active_support/cache/memory_store.rb:
```diff
@@ -4,7 +4,7 @@ module ActiveSupport
   module Cache
     # A cache store implementation which stores everything into memory in the
     # same process. If you're running multiple Ruby on Rails server processes
-    # (which is the case if you're using mongrel_cluster or Phusion Passenger),
```
...if you're using Phusion Passenger or puma clustered mode)

vipulnsward commented on activesupport/lib/active_support/backtrace_cleaner.rb:
```diff
@@ -59,8 +59,8 @@ def add_filter(&block)
     #
-    #   # Will reject all lines that include the word "mongrel", like "/gems/mongrel/server.rb" or "/app/my_mongrel_server/rb"
-    #   backtrace_cleaner.add_silencer { |line| line =~ /mongrel/ }
+    #   # Will reject all lines that include the word "awesome", like "/libs/awesome/server.rb" or "/app/my_awesome_library.rb"
```
same here

vipulnsward commented on guides/source/caching_with_rails.md:
```diff
@@ -381,7 +381,7 @@ config.cache_store = :memory_store, { size: 64.megabytes }
 
 If you're running multiple Ruby on Rails server processes (which is the case
-if you're using mongrel_cluster or Phusion Passenger), then your Rails server
+if you're using Phusion Passenger), then your Rails server
```
ditto

guilleiguaran commented:
+1 for changing this from mongrel to puma instead of using fake names.

tricknotes committed:
Remove the word "mongrel" from documents

Currently mongrel is not maintained.
And it couldn't be built with any Ruby versions that
supported by Rails.

It is reasonable to remove the word "mongrel" in order to avoid
confusion from newcomer.

tricknotes commented:
@vipulnsward @guilleiguaran 
Thanks for your review :-)
I agreed with your comments and force pushed with these changes.

tricknotes commented:
@vipulnsward 
I see.
I'll add this one in next time :-)
